src/stores/Create.ts

```diff
@@ -145,32 +153,41 @@ export const getInitialStateData = (): Pick<
     | 'endpointSchema'
     | 'collections'
     | 'resourceConfig'
+    | 'messagePrefix'
 > => {
     return {
+        messagePrefix,
         details: initialCreateStates.details(),
         endpointConfig: initialCreateStates.endpointConfig(),
         connectors: initialCreateStates.connectors(),
         formState: initialCreateStates.formState(),
         endpointSchema: initialCreateStates.endpointSchema(),
         resourceConfig: initialCreateStates.resourceConfig(),
-        collections: initialCreateStates.collections(),
+        collections: includeCollections
+            ? initialCreateStates.collections()
+            : null,
     };
 };
 
-export const getInitialState = (
+export const getInitialCreateState = (
     set: NamedSet<CreateEntityStore>,
-    get: GetState<CreateEntityStore>
+    get: GetState<CreateEntityStore>,
+    includeCollections: boolean,
+    messagePrefix: Stores
 ): CreateEntityStore => {
     return {
-        ...getInitialStateData(),
+        ...getInitialStateData(includeCollections, messagePrefix),
         setDetails: (details) => {
             set(
                 produce((state) => {
                     if (
                         state.details.data.connectorImage?.id !==
                         details.data.connectorImage?.id
                     ) {
-                        const initState = getInitialStateData();
+                        const initState = getInitialStateData(
```

Comment:
I believe `initState` can be destructured to extract the two properties used in the assignments below. Any reason to not break them out?

Reply:
Updated